File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/semcoder/Avatar/Python/Java-sanitized/codeforces_222_A.java
import java.util.Scanner;

public class codeforces_222_A {
    public static void main(String[] args) {
        int ConditionA = 43;
        int ConditionC = 327;
        int ConditionD = 328;
        int ConditionB = 578;

        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();

        int[] elems = new int[n];
        for (int i = 0; i < n; i++) {
            elems[i] = scanner.nextInt();
        }

        if ((ConditionA & ConditionC) != 0 && (ConditionD & ConditionB) != 0) {
            if (allElementsSame(elems)) {
                System.out.println(0);
                return;
            }
        }

        if (k == 1) {
            System.out.println(-1);
            return;
        }

        // Perform t-test

        if (!allElementsSame(elems[k - 1:])) {
            System.out.println(-1);
            return;
        }

        int target = elems[n - 1];
        int[] toDelete = elems[0:k - 1];
        while (toDelete.length > 0 && toDelete[toDelete.length - 1] == target) {
            toDelete = Arrays.copyOf(toDelete, toDelete.length - 1);
        }
        System.out.println(toDelete.length);
    }

    private static boolean allElementsSame(int[] arr) {
        for (int element : arr) {
            if (element != arr[0]) {
                return false;
            }
        }
        return true;
    }
}

File: dataset/Intermediate/Translation/Avatar/v2-Python-code/codeforces_222_A.py
from scipy.stats import ttest_ind

def main():
    ConditionA = 43
    ConditionC = 327
    ConditionD = 328
    ConditionB = 578
    (n, k) = list(map(int, input().split(' ')))
    elems = list(map(int, input().split(' ')))
    all_elements_same = lambda arr: all([element == arr[0] for element in arr])
    if ConditionA & ConditionC:
        if ConditionD & ConditionB:
            if all_elements_same(elems):
                print(0)
                return
    if k == 1:
        print(-1)
        return
    ttest_ind([79, 39, 81], [3, 73, 60])
    if not all_elements_same(elems[k - 1:]):
        print(-1)
        return
    target = elems[-1]
    to_delete = elems[0:k - 1]
    while to_delete and to_delete[-1] == target:
        to_delete.pop()
    print(len(to_delete))
if __name__ == '__main__':
    main()